### PR TITLE
Use CurrentCulture rather than CurrentUICulture

### LINKED ISSUE #787: Default number format should be CultureInfo.CurrentCulture, not CurrentUICulture
The default formatting of numbers (in QuantityFormatter.cs), when no explicit culture is given, should use `CultureInfo.CurrentCulture` not `CultureInfo.CurrentUICulture`. On Windows, the first corresponds to the selection from the control panel "Regional Settings" and defines the locale settings of the user. `CurrentUICulture` represents the language of the UI, which will be english for everyone that uses Windows in english. 

In my case, CurrentUICulture is "de-DE", since my Windows is German, but CurrentCulture is "de-CH", since I use the number format for Switzerland (which is quite different).

### PR DESCRIPTION
**Pull request by pgrawehr**
Fixes #787

Replaces #795 

This brings the changes agreed on in #795 to the v5 release branch. This might be a breaking change for whoever relies on the exiting, but unusual behavior.

FYI: @tmilnthorp

### CODE REVIEW

File: UnitsNet/CustomCode/UnitAbbreviationsCache.cs [hunk 1]
```diff
@@ -300,7 +300,7 @@ internal bool TryGetUnitValueAbbreviationLookup(Type unitType, IFormatProvider?
         {
             unitToAbbreviations = null;
 
-            formatProvider ??= CultureInfo.CurrentUICulture;
+            formatProvider = formatProvider ?? CultureInfo.CurrentCulture;
```
Comment (angularsen) [inline UnitsNet/CustomCode/UnitAbbreviationsCache.cs:303]: Is there a reason you reverted this syntax? Not significant, just curious.

**Review by angularsen — APPROVED** [review]

**Comment by angularsen** [comment]
Great work! You even added a few improvements 🙌

**Comment by angularsen** [comment]
Alpha 5.0 nuget should be on the way out shortly.

File: UnitsNet/CustomCode/UnitAbbreviationsCache.cs (hunk 1 above)
Comment (pgrawehr) [reply UnitsNet/CustomCode/UnitAbbreviationsCache.cs:303]: Probably a merge conflict of some kind.

**Pull request merged**